antora/supplemental-ui/helpers/parseHugoMenu.js

```diff
@@ -0,0 +1,46 @@
+const fs = require("fs");
+const toml = require("toml");
+
+module.exports = () => {
+  const filePath = "./hugo.toml";
+  const fileContent = fs.readFileSync(filePath, "utf8");
+  const config = toml.parse(fileContent);
+  const menu = config.menu.main || [];
+  const processed = new Set();
+
+  const createMenu = (item) => {
+    if (processed.has(item)) {
+      return {};
+    }
+    processed.add(item);
+    const children = menu
+      .filter((child) => child.parent === item.name)
+      .map(createMenu);
+    if (children.length) {
+      return {
+        name: item.name,
+        url: item.url,
+        pageRef: item.pageRef,
+        weight: item.weight || 0,
+        parent: item.parent,
+        graphic: item.graphic,
+        children,
+      };
+    } else {
+      return {
+        name: item.name,
```

Comment:
NIT: It would be nice to not have to duplicate the whole object:

```js
const menuItem = {
        name: item.name,
        url: item.url,
        pageRef: item.pageRef,
        weight: item.weight || 0,
        parent: item.parent,
        graphic: item.graphic,
}
if (children.length) {
  menuItem["children"] = children
}
return menuItem
```